File: client-scalaz/src/main/scala/org/openapitools/client/api/LocationResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import LocationResponse._

case class LocationResponse (
  valid: Option[Boolean],
message: Option[String],
version: Option[Double],
serializeNulls: Option[Boolean],
startTimeLog: Option[Long],
errorCode: Option[String],
request: Option[List[NameStringValueResponse]],
id: Option[Long],
getDistanceMiles: Option[String],
zip: Option[String],
name: Option[String],
latitude: Option[String],
longitude: Option[String],
getSource: Option[String],
`type`: Option[String],
returning: Option[String])

object LocationResponse {
  import DateTimeCodecs._

  implicit val LocationResponseCodecJson: CodecJson[LocationResponse] = CodecJson.derive[LocationResponse]
  implicit val LocationResponseDecoder: EntityDecoder[LocationResponse] = jsonOf[LocationResponse]
  implicit val LocationResponseEncoder: EntityEncoder[LocationResponse] = jsonEncoderOf[LocationResponse]
}

File: client-clojure/src/sirqul_io_t_platform/specs/location_response.clj
(ns sirqul-io-t-platform.specs.location-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.name-string-value-response :refer :all]
            )
  (:import (java.io File)))


(def location-response-data
  {
   (ds/opt :valid) boolean?
   (ds/opt :message) string?
   (ds/opt :version) float?
   (ds/opt :serializeNulls) boolean?
   (ds/opt :startTimeLog) int?
   (ds/opt :errorCode) string?
   (ds/opt :request) (s/coll-of name-string-value-response-spec)
   (ds/opt :id) int?
   (ds/opt :get_distance_miles) string?
   (ds/opt :zip) string?
   (ds/opt :name) string?
   (ds/opt :latitude) string?
   (ds/opt :longitude) string?
   (ds/opt :get_source) string?
   (ds/opt :type) string?
   (ds/opt :returning) string?
   })

(def location-response-spec
  (ds/spec
    {:name ::location-response
     :spec location-response-data}))
